core/src/main/java/org/elasticsearch/cluster/routing/RoutingService.java

```diff
@@ -62,6 +65,10 @@
     private volatile boolean routingTableDirty = false;
 
     private volatile Future scheduledRoutingTableFuture;
+    private AtomicBoolean rerouting = new AtomicBoolean();
+
+    private volatile long registeredNextDelaySetting = Long.MAX_VALUE;
+    private volatile ScheduledFuture registeredNextDelayFuture;
```

Comment:
its sneaky, conceptually, it doesn't, due to the fact that we have single threaded cluster service execution, I think its more readable that way, so we don't worry about it